src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -703,6 +703,11 @@ private BoundExpression CreateAnonymousFunctionConversion(SyntaxNode syntax, Bou
             var boundLambda = unboundLambda.Bind((NamedTypeSymbol)destination, isExpressionTree: destination.IsGenericOrNonGenericExpressionType(out _));
             diagnostics.AddRange(boundLambda.Diagnostics);
 
+            if (conversion.HasWarning)
```

Comment:
>conversion.HasWarning [](http://example.com/codeflow?start=16&length=21)

Does `GenerateImplicitConversionError()` report a warning in this case? And if `Conversion` has a `HasWarning` property, does that mean that all code paths that create `Conversion` instances need to be aware of this possibility? If this is specific to the anonymous function conversion code path, perhaps it should be separated out from `Conversion` or explicitly named.

Reply:
Yes, `GenerateImplicitConversionError` would report a warning in this case, though not for anonymous functions. I originally added the `HasWarning` property only for the anonymous function conversion code path, but it is no longer needed. It will be needed for method groups however, so I could rename to something like `HasMethodGroupConversionWarning`?

Reply:
Attaching that field to `Conversion` really did seem like the simplest way to go about this, rather than trying propagate a separate `hasWarning` parameter

Reply:
If the only warning is specific to anonymous functions or method groups and only handled here, and if the check is simple, we should consider simply moving the check and the error reporting here rather than adding to `Conversion` and `GenerateImplicitConversionError()`.

Reply:
I think I will remove this check (and the `HasWarning` field) for now, and address this in my next PR for method groups